rfcs/framework-0072-pluggable-transfers.md

```diff
@@ -0,0 +1,507 @@
+[AWS S3 Multipart](https://www.npmjs.com/package/@uppy/aws-s3-multipart) package, adapted to match REST APIs designed by this RFC by the [InvenioMultipartUploader](https://github.com/oarepo/invenio-files-uppy-js/blob/main/src/components/UppyUploader/InvenioMultipartUploader.js) Uppy plugin.
+
+Drag & drop file input with auto-thumbnails, upload progress tracking, upload abortion & retry and status bar is provided out of the box. Files Redux store actions is still being used to manage record files state.
+
+More features & integrations like the remote file sources, image or file metadata editor, could be potentially added to the Dashboard UI later.
+
+#### Usage
+
+Uppy integration could be enabled by registering it from a [Webpack Files UI entrypoint](#webpack-files-ui-entrypoint) and then setting it to be used from the site config:
+```python
+FILES_UI_ENTRYPOINT = "invenio-app-rdm-files-uppy"
+```
+
+#### UI workflow
+
+Default dashboard state:
+![Screenshot 2025-02-13 at 8 57 20](https://github.com/user-attachments/assets/85319625-7510-4482-b314-64eac835fc73)
+
+Files chosen for upload:
+![Screenshot 2025-02-13 at 9 07 20](https://github.com/user-attachments/assets/eb1810d5-fd27-4612-9de7-ad028c78c1ca)
+
+Multipart upload in progress:
+![Screenshot 2025-02-13 at 9 07 38](https://github.com/user-attachments/assets/c7be5987-7848-4322-9baf-af8cd9a28e9b)
+
+Upload complete:
+![Screenshot 2025-02-13 at 9 07 55](https://github.com/user-attachments/assets/a5820012-a16a-4270-97ee-82d1392ad948)
+
+Add one more file that fails to upload:
+![Screenshot 2025-02-13 at 9 08 58](https://github.com/user-attachments/assets/252826f3-fb66-44b2-a211-4a72bb0d2564)
+
+## Other Examples
+
+Another potential use case is a specialized transport for handling
+sensitive data in the Crypt4GH file format before storing it in the
+repository (managing encryption keys, etc.).
+
+## How we teach this
+
+A [file storage documentation](https://github.com/inveniosoftware/docs-invenio-rdm/blob/master/docs/reference/file_storage.md)
+page will be updated to reflect these API changes.
+
+Back-end part is primarily
+an internal feature, detailed implementation notes will remain in this RFC
+and in the API documentation.
+
+On front-end:
+- For developers, a documentation on how to build a custom Files UI plugin is probably desirable.
+- For site admins, we need to document the `FILES_UI_ENTRYPOINT` config option.
+
+## Drawbacks
+
+- A slightly more complicated code.
+- Some files UI components and Redux store actions that were private now need to be exported to allow reuse.
+- Reliance on the global state (`window.invenio.files`) for registering the Files UI, along with an additional Webpack entry point that must be included in a specific order.
+- Need to create a Webpack entrypoint in site's webpack to be able to use any custom Files UI implementations, that are just NPM packages (like the Uppy.io integration above).
```

Comment:
Not sure if it's intended or not, but I feel that the Uppy.io component and Webpack entrypoint should/can already live alongside the current file uploader component, as a first-class supported option for InvenioRDM.